Name: Temperature with Battery in Device Menu
Version: 1.4.0-1
Author: Jason Robitaille
Description: Will add a temperature value on the right side of the battery row of the device menu. Will not conflict with any other device menu patches.  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -112,6 +112,12 @@
 			this.controller.hideWidgetContainer(this.controller.get('dm_wifi'));
 		}
 		
+		//setup battery temperature label
+		var battery = this.controller.get("dm_battery");
+		battery.id = "batteryRow";
+		battery.innerHTML = '<span id="dm_battery">' + battery.innerHTML + '</span>' +
+				'<div class="label right" id="tempmsg" style="font-size:0.9em;">&nbsp;</div>';
+		
 		//set up the the menu div
 		this.scrim = this.controller.sceneElement.querySelector('div[x-mojo-scrim]');
 		this.systemmenu = this.controller.sceneElement.querySelector('div[x-mojo-dialog]');
@@ -1078,6 +1084,21 @@
 				this.btspinnerModel.spinning = true;
 				this.controller.modelChanged(this.btspinnerModel);
 			}
+			this.controller.serviceRequest('palm://com.palm.bus/signal/', {
+				method: 'addmatch',
+				parameters: {
+					"category":"/com/palm/power",
+					"method":"batteryStatus"
+				},
+				onSuccess: function(payload) {
+					if(payload.temperature_C) {
+						this.controller.get("tempmsg").innerHTML = payload.temperature_C + "&deg;C";
+					}
+				}.bind(this)
+			});
+			this.controller.serviceRequest('palm://com.palm.power/com/palm/power/', {
+				method: 'batteryStatusQuery'
+			});
 			this.controller.stageController.activate();
 			
 			var animateDialog = function(){
